doc/neps/indexing.rst

```diff
@@ -0,0 +1,506 @@
+``arr.oindex``, ``arr.vindex``, and ``arr.lindex`` to allow these indexing
+operations. Also starting to deprecate those plain index operations
+which are not ambiguous.
+Furthermore, the NumPy code base will need to use the new attributes and
+tests will have to be adapted.
+
+
+Alternative Names
+=================
+
+Possible names suggested (more suggestions will be added).
+
+==============  ======== ======= ============
+**Orthogonal**  oindex   oix
+**Vectorized**  vindex   fix
+**Legacy**      l/findex         legacy_index
+==============  ======== ======= ============
+
+
+Subclasses
+==========
+
+Subclasses are a bit problematic in the light of these changes. There are
+some possible solutions for this. For most subclasses (those which do not
+provide ``__getitem__`` or ``__setitem__``) the special attributes should
+just work. Subclasses that *do* provide it must be updated accordingly
+and should preferably not subclass working versions of these attributes.
+
+All subclasses will inherit the attributes, however, it seems possible
+to test ``subclass.__getitem__.__classobj__`` when getting i.e.
+``subclass.vindex``. If this is not ``ndarray``, the subclass has special
+handling and an ``AttributeError`` can be given.
+
+A further question is how to facilitate implementing the special attributes.
+Also there is the weird functionality where ``__setitem__`` calls
+``__getitem__`` for non-advanced indices. It might be good to avoid it for
+the new attributes, but on the other hand, that may make it even more
+confusing.
+
+To facilitate implementations we could provide functions similar to
+``operator.itemgetter`` and ``operator.setitem`` for the attributes.
+Possibly a MixIn could be provided to help implementation, but further
+ideas on this are necessary.
+
+
+Examples
+========
+
+Since the various kinds of indexing is hard to grasp in many cases, these
+examples hopefully give some more insights. Note that they are all in terms
+of shape.
+In the examples, all original dimensions have 5 or more elements,
+advanced indexing inserts smaller dimensions.
+These examples may be hard to grasp without working knowledge of advanced
+indexing as of NumPy 1.9.
+
+Example array::
+
+    >>> arr = np.ones((5, 6, 7, 8))
+
+
+Legacy fancy indexing
+---------------------
+
+Note that the same result can be achieved with ``arr.lindex``, but the
+"future error" will still work in this case.
+
+Single index is transposed (this is the same for all indexing types)::
+
+    >>> arr[[0], ...].shape
+    (1, 6, 7, 8)
+    >>> arr[:, [0], ...].shape
+    (5, 1, 7, 8)
+
+
+Multiple indices are transposed *if* consecutive::
+
+    >>> arr[:, [0], [0], :].shape  # future error
+    (5, 1, 7)
+    >>> arr[:, [0], :, [0]].shape  # future error
+    (1, 5, 6)
+
+
+It is important to note that a scalar *is* integer array index in this sense
+(and gets broadcasted with the other advanced index)::
+
+    >>> arr[:, [0], 0, :].shape
+    (5, 1, 7)
```

Comment:
Yes, you were right of course. There was a typo. Intuition is such a nasty thing though, I hope that it becomes a little clear why it is like that currently, though I guess it is mostly one of the big reasons in favor if forcing things like `oindex`.